src/gateways/pedidoGateway.ts

```diff
@@ -185,6 +185,33 @@ export class PedidoMongoGateway implements PedidoGateway {
         );
     }
 
+    async updateStatus(
+        id: string,
+        pedido: Omit<Partial<Pedido>, "id" | "cliente">,
+    ): Promise<Pedido> {
+        const result = await this.pedidoModel
+            .findOneAndUpdate({ _id: id }, pedido, {
+                new: true,
+            })
+            .populate<{ cliente: ClienteDTO }>("cliente");
+
+        return PedidoMapper.toDomain(
+            {
+                id: result.id,
+                status: result.status,
+                valorTotal: result.valorTotal,
+                itens: result.itens,
+                observacoes: result.observacoes,
+            },
```

Comment:
Eu vou comentar nessa linha, mas seria interessante adicionar pra todos os métodos do gateway de pedidos. No meio dos testes que eu estava fazendo, mesmo atualizando no banco o status de pagamento não saia de pendente. 

Ai dando uma olhada, eu descobri que é porque o mapper do pedido (no gateway),  acabou ficando sem receber a propriedade de pagamento, pra mostrar o status atualizado.

Ai pra consertar seria só dar uma passada nos métodos do gateway e adicionar o mapping pra o status de pedidos, exemplo:

```suggestion
            {
                id: result.id,
                status: result.status,
                pagamento: result.pagamento,
                valorTotal: result.valorTotal,
                itens: result.itens,
                observacoes: result.observacoes,
            },
```